From ef1a5870aae22df707f0f25e83bb332441cf31e7 Mon Sep 17 00:00:00 2001
From: William Blake Galbreath <Blake.Galbreath@GMail.com>
Date: Tue, 9 Jun 2020 22:46:29 -0500
Subject: [PATCH] Configurable custom server brand


diff --git a/proxy/src/main/java/net/md_5/bungee/ServerConnector.java b/proxy/src/main/java/net/md_5/bungee/ServerConnector.java
index 35a19224..c8def9d5 100644
--- a/proxy/src/main/java/net/md_5/bungee/ServerConnector.java
+++ b/proxy/src/main/java/net/md_5/bungee/ServerConnector.java
@@ -250,8 +250,15 @@ public class ServerConnector extends PacketHandler
 
             user.unsafe().sendPacket( modLogin );
 
+            // Pl3x start
+            String displayBrand = ProxyServer.getInstance().getTranslation("custom_brand");
+            if (displayBrand == null || displayBrand.isEmpty()) {
+                displayBrand = bungee.getName() + " (" + bungee.getVersion() + ")";
+            }
+            // Pl3x end
+
             ByteBuf brand = ByteBufAllocator.DEFAULT.heapBuffer();
-            DefinedPacket.writeString( bungee.getName() + " (" + bungee.getVersion() + ")", brand );
+            DefinedPacket.writeString( displayBrand, brand ); // Pl3x
             user.unsafe().sendPacket( new PluginMessage( user.getPendingConnection().getVersion() >= ProtocolConstants.MINECRAFT_1_13 ? "minecraft:brand" : "MC|Brand", brand, handshakeHandler.isServerForge() ) );
             brand.release();
 
diff --git a/proxy/src/main/java/net/md_5/bungee/connection/DownstreamBridge.java b/proxy/src/main/java/net/md_5/bungee/connection/DownstreamBridge.java
index d7d0fafc..d115f6d8 100644
--- a/proxy/src/main/java/net/md_5/bungee/connection/DownstreamBridge.java
+++ b/proxy/src/main/java/net/md_5/bungee/connection/DownstreamBridge.java
@@ -280,8 +280,15 @@ public class DownstreamBridge extends PacketHandler
 
             Preconditions.checkState( !serverBrand.contains( bungee.getName() ), "Cannot connect proxy to itself!" );
 
+            // Pl3x start
+            String displayBrand = ProxyServer.getInstance().getTranslation("custom_brand");
+            if (displayBrand == null || displayBrand.isEmpty()) {
+                displayBrand = bungee.getName() + " (" + bungee.getVersion() + ")" + " <- " + serverBrand;
+            }
+            // Pl3x end
+
             brand = ByteBufAllocator.DEFAULT.heapBuffer();
-            DefinedPacket.writeString( bungee.getName() + " (" + bungee.getVersion() + ")" + " <- " + serverBrand, brand );
+            DefinedPacket.writeString( displayBrand, brand ); // Pl3x
             pluginMessage.setData( brand );
             brand.release();
             // changes in the packet are ignored so we need to send it manually
diff --git a/proxy/src/main/java/net/md_5/bungee/connection/InitialHandler.java b/proxy/src/main/java/net/md_5/bungee/connection/InitialHandler.java
index 37ea35e3..d99cc067 100644
--- a/proxy/src/main/java/net/md_5/bungee/connection/InitialHandler.java
+++ b/proxy/src/main/java/net/md_5/bungee/connection/InitialHandler.java
@@ -173,7 +173,14 @@ public class InitialHandler extends PacketHandler implements PendingConnection
         this.legacy = true;
         final boolean v1_5 = ping.isV1_5();
 
-        ServerPing legacy = new ServerPing( new ServerPing.Protocol( bungee.getName() + " " + bungee.getGameVersion(), bungee.getProtocolVersion() ),
+        // Pl3x start
+        String displayBrand = net.md_5.bungee.api.ProxyServer.getInstance().getTranslation("custom_brand");
+        if (displayBrand == null || displayBrand.isEmpty()) {
+            displayBrand = bungee.getName() + " " + bungee.getGameVersion();
+        }
+        // Pl3x end
+
+        ServerPing legacy = new ServerPing( new ServerPing.Protocol( displayBrand, bungee.getProtocolVersion() ), // Pl3x
                 new ServerPing.Players( listener.getMaxPlayers(), bungee.getOnlineCount(), null ),
                 new TextComponent( TextComponent.fromLegacyText( listener.getMotd() ) ), (Favicon) null );
 
@@ -221,8 +228,15 @@ public class InitialHandler extends PacketHandler implements PendingConnection
 
     private ServerPing getPingInfo(String motd, int protocol)
     {
+        // Pl3x start
+        String displayBrand = net.md_5.bungee.api.ProxyServer.getInstance().getTranslation("custom_brand");
+        if (displayBrand == null || displayBrand.isEmpty()) {
+            displayBrand = bungee.getName() + " " + bungee.getGameVersion();
+        }
+        // Pl3x end
+
         return new ServerPing(
-                new ServerPing.Protocol( bungee.getName() + " " + bungee.getGameVersion(), protocol ),
+                new ServerPing.Protocol( displayBrand, protocol ), // Pl3x
                 new ServerPing.Players( listener.getMaxPlayers(), bungee.getOnlineCount(), null ),
                 motd, BungeeCord.getInstance().config.getFaviconObject()
         );
diff --git a/proxy/src/main/resources/messages.properties b/proxy/src/main/resources/messages.properties
index 6e3ef0a4..63a6b484 100644
--- a/proxy/src/main/resources/messages.properties
+++ b/proxy/src/main/resources/messages.properties
@@ -38,3 +38,4 @@ you_got_summoned=\u00a76Summoned to {0} by {1}
 command_perms_groups=\u00a76You have the following groups: {0}
 command_perms_permission=\u00a79- {0}
 command_ip=\u00a79IP of {0} is {1}
+custom_brand=
-- 
2.26.2

